src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Cookies.cs

```diff
@@ -279,9 +256,23 @@ public async Task GetAsync_SetCookieContainerAndMultipleCookieHeaders_BothCookie
                     await TestHelper.WhenAllCompletedOrAnyFailed(getResponseTask, serverTask);
 
                     HttpRequestData requestData = await serverTask;
-                    string cookieHeaderValue = requestData.GetSingleHeaderValue("Cookie");
+                    string[] cookieHeaderValues = requestData.GetHeaderValues("Cookie");
+                    string cookieHeaderValue = null;
 
-                    // Multiple Cookie header values are treated as any other header values and are 
+                    foreach (string header in cookieHeaderValues)
```

Comment:
Since this logic is duplicated, it's probably worth separating out into a separate function -- especially once you add the additional logic I suggest above